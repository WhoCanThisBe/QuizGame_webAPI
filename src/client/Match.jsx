import {getRandomQuizzes} from "../server/db/quizzes";
import React from "react";

export function Match() {
    const currentQuiz = getRandomQuizzes(1)[0];

    function handleClick(correct) {
        console.log(correct);

        if (!correct) return alert("wrong");

        alert("correct");

    }

    return (
        <div data-testid={"questions"} className='question'>
            <h1>Question (1 / 3): {currentQuiz.question}</h1>
            {currentQuiz.answers.map((alternative, index) => (
                <button
                    type="button"
                    key={index}
                    onClick={() =>
                        handleClick(index === currentQuiz.indexOfRightAnswer)
                    }
                >
                    {alternative}
                </button>
            ))}
        </div>
    );
}