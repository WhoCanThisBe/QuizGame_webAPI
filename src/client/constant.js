
export  const NAV_PATH ={
  HOME: "/"
}