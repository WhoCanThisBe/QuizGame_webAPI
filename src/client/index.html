<!DOCTYPE html>
<html>
<head>
    <meta charset="UTF-8">
    <title>Quiz Game</title>
    <link rel="stylesheet" href="style.css"/>
</head>
<body>
<div class="container">
    <h1 class="heading">Quiz Game</h1>

    <div class="main-content" id="quizDivId" />
</div>

<script src="code.js"></script>
<script>
    displayNewQuiz();
</script>

</body>
</html>
