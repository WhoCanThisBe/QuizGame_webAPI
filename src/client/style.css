.container {
  display: flex;
  flex-direction: column;
  align-content: center;
  align-items: center;
}

.main-content {
  display: flex;
  flex-direction: column;
  min-width: 50%;
  align-items: center;
}

.heading {
  font-size: 4em;
  margin: 2rem 0;
}

.welcome-text {
  font-size: 1.5em;
  margin-bottom: 1rem;
}
.action {
  padding: 1rem 0;
}

.play {
  background-color: #ebe9e4;
  color: #000000;
  padding: 1rem;
  border: 1px solid #787774;
  border-radius: 5px;
  margin-top: 1rem;
}

.play:focus,
.play:hover {
  background-color: #2d1ed6;
  color: #ebe9e4;
  /*
    The focus selector is important for universal design.
    Especially for users navigating with keyboard. Sadly it is often forgotten
    and a lot of designers request frontend developers to remove the browsers default focus,
    without adding a custom one. This should not be done!
    If you do this you make it very hard for a keyboard user to navigate.
    The least you can do is to leave the Default browser style and then add the same style as on hover.
    Read more: https://www.deque.com/blog/give-site-focus-tips-designing-usable-focus-indicators/
    */
}

.quiz {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.question {
  font-size: 1.5em;
  font-weight: bold;
}

.answer {
  background-color: #ebe9e4;
  margin-bottom: 1rem;
  padding: 1rem;
  border: 1px solid #787774;
  border-radius: 5px;
  width: 80%;
}

.answer:hover,
.answer:focus {
  background-color: #2d1ed6;
  color: #ebe9e4;
}

.game-result {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.new-game-button {
  align-self: center;
}
