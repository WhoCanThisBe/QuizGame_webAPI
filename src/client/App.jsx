import React from 'react';
import {BrowserRouter as Router, Route, Switch} from "react-router-dom";
import {NAV_PATH} from "./constant";
import {Match} from "./Match";


const App = ()  => {
    return (
        <Router>
            <Switch>
                <Route exact path = {NAV_PATH.HOME}>
                    <Match/>
                </Route>
            </Switch>
        </Router>
    );
};

App.propTypes = {

};

export default App;