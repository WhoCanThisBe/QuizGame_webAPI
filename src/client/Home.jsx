import { Link } from "react-router-dom";
import React from "react";
import { NAV_PATH } from "./constant";

export function Home() {
  return (
    <div>
      <h2>Quiz Game</h2>

      <p className={"welcome-text"}>
        Welcome to the Quiz Game! In this game, you will get a series of
        questions, each one with 4 possible answers. Only 1 out of the 4 answers
        is correct. If you answer wrongly to any of the questions, you lose! You
        win if you manage to answer correctly to all questions.
      </p>

      <div className={"action"}>
        <Link to={NAV_PATH.MATCH} className={"play"}>
          Play
        </Link>
      </div>
    </div>
  );
}
