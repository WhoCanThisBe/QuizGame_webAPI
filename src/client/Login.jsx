import React, { useState } from "react";
import { StatusCode } from "status-code-enum";
import { NAV_PATH, USER_AUTH_ENDPOINT } from "./constant";
import { useHistory } from "react-router";
import { useSubmit } from "./customhooks/useSubmit";
import { post, postJSON } from "./lib/http";

const Login = ({ setLoggedIn }) => {
  const [userId, setUserId] = useState("");
  const [password, setPassword] = useState("");
  const history = useHistory();

  const { handleSubmit, submitting, error } = useSubmit(
    () => post(USER_AUTH_ENDPOINT.LOGIN, { userId, password }),
    () => {
      setLoggedIn();
      history.push(NAV_PATH.HOME);
    }
  );

  const errorMsg =
    error?.status === StatusCode.ClientErrorUnauthorized
      ? "Invalid username/password"
      : error?.toString();

  return (
    <div className="center">
      <form onSubmit={handleSubmit}>
        <h1>Login</h1>
        <label>
          User Id:
          <br />
          <input
            type="text"
            value={userId}
            onChange={(e) => setUserId(e.target.value)}
            required
          />
        </label>
        <br />
        <br />
        <label>
          Password:
          <br />
          <input
            type="password"
            value={password}
            onChange={(e) => setPassword(e.target.value)}
            required
          />
        </label>
        <br />
        <br />
        {error && <p>{errorMsg}</p>}

        {/*So no one can submit while one is submitting*/}
        <button
          className="button"
          type={"submit"}
          disabled={!password || submitting}
        >
          Login
        </button>
      </form>
    </div>
  );
};

export default Login;
