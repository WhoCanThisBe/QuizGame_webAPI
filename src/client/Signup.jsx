import React, { useState } from "react";
import { useHistory } from "react-router";
import { post, postJSON } from "./lib/http";
import { NAV_PATH, USER_AUTH_ENDPOINT } from "./constant";
import { useSubmit } from "./customhooks/useSubmit";

const Signup = ({ setLoggedIn, ...props }) => {
  const [userId, setUserId] = useState("");
  const [password, setPassword] = useState("");
  const [confirm, onConfirmChange] = useState("");
  const history = useHistory();

  const { handleSubmit, submitting, error } = useSubmit(
    () => post(USER_AUTH_ENDPOINT.SIGNUP, { userId, password }),
    () => {
      // TODO: Remove/change this after setting up WebSockets
      // Trigger fetching of userInfo in App
      setLoggedIn();
      history.push(NAV_PATH.HOME);
    }
  );

  const isTheSame = password === confirm;

  let confirmMsg = isTheSame ? "ok" : "not the same";

  return (
    <div className="center">
      <form onSubmit={handleSubmit}>
        <h1>Signup</h1>
        <label>
          User Id:
          <br />
          <input
            type="text"
            value={userId}
            onChange={(e) => setUserId(e.target.value)}
            required
          />
        </label>
        <br />
        <br />
        <label>
          Password:
          <br />
          <input
            type="password"
            value={password}
            onChange={(e) => setPassword(e.target.value)}
            required
          />
        </label>
        <br />
        <br />
        <label>
          Confirm:
          <br />
          <input
            type="password"
            value={confirm}
            onChange={(e) => onConfirmChange(e.target.value)}
            required
          />
          <div>{confirm.length > 0 && confirmMsg}</div>
          <br />
        </label>

        {error && <p>{error.toString()}</p>}

        <button
          className="button"
          type={"submit"}
          disabled={!isTheSame || submitting}
        >
          Sign Up
        </button>
      </form>
    </div>
  );
};

export default Signup;
