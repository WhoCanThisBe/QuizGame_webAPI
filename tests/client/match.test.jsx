import React from "react";
import {Match} from "../../src/client/Match";
import {MemoryRouter} from "react-router-dom";
import renderer from 'react-test-renderer';


test.skip("display quiz",() =>{

    //TODO: make an better test

    const matchDisplay = (
        <MemoryRouter>
            <Match />
        </MemoryRouter>
    );

    const tree = renderer.create(matchDisplay).toJSON();

    expect(tree).toMatchSnapshot();

})




