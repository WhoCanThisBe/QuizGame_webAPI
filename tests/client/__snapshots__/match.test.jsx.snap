// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display quiz 1`] = `
<div
  className="question"
  data-testid="questions"
>
  <h1>
    Question (1 / 3): 
    In JavaScript, what is the result  of the following?

false + true?
  </h1>
  <button
    onClick={[Function]}
    type="button"
  >
    false
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    true
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    'falsetrue'
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    1
  </button>
</div>
`;
